### PR TITLE
feat(experiment): "Make decision" follow ups

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Changes
- add the ability to stop the experiment from the Make decision modal
- unit test the method for transforming feature flag filters when shipping the variant
- add event tracking
- fix a bug in ResetButton which was causing the confirmation dialog to pop up twice

https://github.com/user-attachments/assets/644da0a6-5d4c-4db3-a858-8584230c92c8

## How did you test this code?
test, 👀

### CODE REVIEW

File: frontend/src/scenes/experiments/ExperimentView/components.tsx [hunk 1]
```diff
@@ -542,34 +516,55 @@ export function MakeDecisionModal({ experimentId }: { experimentId: Experiment['
...
+                            }))}
+                        />
+                    </div>
+                    {!isExperimentStopped && (
+                        <>
+                            <LemonDivider className="my-0" vertical />
+                            <div className="w-2/5 pl-4">
+                                <LemonCheckbox
+                                    id="flag-enabled-checkbox"
+                                    label="Stop experiment"
```
Comment (dmarticus) [inline frontend/src/scenes/experiments/ExperimentView/components.tsx:560]: can we get a tooltip on this?  Something that, idk, explains why this checkbox exists at all, and the ramifications of stopping the experiment when shipping the variant vs keeping it going?  As a user, it's not immediately clear what checking/unchecking this box does within the modal context.  Tooltips are unreasonably effective and providing explanations without cluttering visuals, so IMO it feels like it'd be a nice lift to include.

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 2]
```diff
@@ -1067,6 +1068,15 @@ export const eventUsageLogic = kea<eventUsageLogicType>([
...
         },
+        reportExperimentVariantShipped: ({ experiment }) => {
+            posthog.capture('experiment variant shipped', {
+                name: experiment.name,
+                id: experiment.id,
+                filters: sanitizeFilterParams(experiment.filters),
+                parameters: experiment.parameters,
+                secondary_metrics_count: experiment.secondary_metrics.length,
+            })
+        },
```
Comment (dmarticus) [inline frontend/src/lib/utils/eventUsageLogic.ts:1079]: image

**Review by dmarticus — APPROVED** [review]
Looks great, nice work man.  Left a note on wanting a tooltip on the checkbox, but this is otherwise excellent.

File: frontend/src/scenes/experiments/ExperimentView/components.tsx (hunk 1 above)
Comment (jurajmajerik) [reply frontend/src/scenes/experiments/ExperimentView/components.tsx]: done! thanks for approving early. I also love tooltips btw :D

**Pull request merged**